### PR TITLE
Adopt typed compare engine output and remove legacy violation contract

### PR DESCRIPTION
**Pull request by corymhall**
## Summary
This PR establishes typed `Change` items as the canonical output of the internal compare engine.

It replaces violation plumbing with a typed-first contract that downstream renderers can consume directly.

## What Changed
- Reworked `internal/compare` model and engine output to emit typed change records.
- Removed legacy violation-oriented report fields and bridging logic.
- Strengthened engine/requiredness tests around deterministic typed output and removed-output edge cases.
- Kept category semantics explicit in one place so later PRs can project/render from a stable contract.

## Why
The stack needs a single canonical change model before projection/rendering and lookup normalization can be simplified safely.

This is the foundation for follow-up projection and lookup integration work in #106 and above.

### CODE REVIEW

File: internal/compare/engine.go [hunk 1]
```diff
@@ -1,31 +1,265 @@
...
+	}
+	if oldType != newType {
+		*changes = append(*changes, newChange(category, name, path, ChangeKindTypeChanged, fmt.Sprintf("type changed from %q to %q", oldType, newType)))
+	}
+
+	appendTypeChanges(changes, category, name, append(slicesClone(path), "items"), old.Items, new.Items)
+	appendTypeChanges(changes, category, name, append(slicesClone(path), "additional properties"), old.AdditionalProperties, new.AdditionalProperties)
+}
+
+func slicesClone(xs []string) []string {
```
Comment (guineveresaenger) [inline internal/compare/engine.go]: Any reason not to use the stdlib https://pkg.go.dev/slices#Clone?

File: internal/compare/engine.go [hunk 2]
```diff
@@ -198,7 +435,7 @@ func BreakingChanges(oldSchema, newSchema schema.PackageSpec) *diagtree.Node {
 		// both inputs and outputs.
 		newRequired := set.FromSlice(newTyp.Required)
 		for _, r := range typ.Required {
-			_, stillExists := typ.Properties[r]
+			_, stillExists := newTyp.Properties[r]
```
Comment (guineveresaenger) [inline internal/compare/engine.go]: same - isn't this an unrelated change?

**Review by guineveresaenger — APPROVED** [review]
I've got a few comments but they may be because I haven't worked my way all the way down the PR stack yet.

File: internal/compare/engine.go (hunk 1 above)
Comment (corymhall) [reply internal/compare/engine.go]: Good catch, I thought I replaced all usages of this type of thing, but missed one!

File: internal/compare/engine.go (hunk 2 above)
Comment (corymhall) [reply internal/compare/engine.go]: Similar reasoning here: this is a bug fix in requiredness gating (`newTyp.Properties` is the right source of truth for “still exists”). Keeping it with this change avoided carrying known-bad behavior into the typed engine transition, and it now has test coverage.
